### PR TITLE
Update frontend node version and migrate configuration from create-react-app to Vite

### PR DESCRIPTION
**Pull request by vejol**
This PR changes frontend's node version from 10 to 20. This is essential for updating other dependencies.

Additionally, configration of the app is migrated from create-react-app to Vite. This will modernize configuration as Vite is recommended for new React apps nowadays. I had to do migration simultaneously with node version change as old create-react-app didn't work with new node and it seemed to be complicated to fix.

I tested new frontend image on staging and everything seemed to work correctly.

### CODE REVIEW

**Review by joonas-somero — APPROVED** [review]

**Review by vejol — COMMENTED** [review]
Good change!

File: frontend/.dockerignore
Comment (Regularmute) [inline frontend/.dockerignore:1]: Missing newline at end.

File: frontend/.gitignore
Comment (Regularmute) [inline frontend/.gitignore:1]: Missing newline at end of file.

File: frontend/Dockerfile
Comment (Regularmute) [inline frontend/Dockerfile:1]: Missing newline at end of file.

File: frontend/index.html
Comment (Regularmute) [inline frontend/index.html:1]: Missing newline at end of file.

**Review by Regularmute — COMMENTED** [review]
Looks good, just tack on the couple missing newlines at the end of some files.

**Review by Regularmute — APPROVED** [review]
Looks good to me.

**Pull request merged**